analyzers/tests/SonarAnalyzer.Test/Rules/BackslashShouldBeAvoidedInAspNetRoutesTest.cs

```diff
@@ -62,4 +62,38 @@ public class BackslashShouldBeAvoidedInAspNetRoutesTest
             .AddReferences(AspNet4xReferences(aspNetMvcVersion))
             .Verify();
 #endif
+
+#if NET
+    public static IEnumerable<object[]> AspNetCore2xVersionsUnderTest => [["2.0.4"], ["2.2.0"], [Constants.NuGetLatestVersion]];
```

Comment:
For my own understanding: How these versions to be tested were picked?

Reply:
Via the `[DynamicData(nameof(AspNetCore2xVersionsUnderTest))]` attribute, used [here](https://github.com/SonarSource/sonar-dotnet/pull/8900/files/ea3911bb3324477e1c39c046d63a9d9c7a352c40#diff-4b54dee07d32e00e08345bd5cd125fe83bc625b350181c3a8b192271c7ff53a5R75), [here](https://github.com/SonarSource/sonar-dotnet/pull/8900/files/ea3911bb3324477e1c39c046d63a9d9c7a352c40#diff-4b54dee07d32e00e08345bd5cd125fe83bc625b350181c3a8b192271c7ff53a5R83) and [here](https://github.com/SonarSource/sonar-dotnet/pull/8900/files/ea3911bb3324477e1c39c046d63a9d9c7a352c40#diff-4b54dee07d32e00e08345bd5cd125fe83bc625b350181c3a8b192271c7ff53a5R92).

When you have multiple parameterized tests, and all need to use the same list of parameters, you can factorize the list of parameters using `DynamicData`. That can also help avoid the "duplicated code" rule.

Reply:
I misunderstood the question.
So the rationale behind the selection of these three versions was the following:
- latest is floating - we should always take it to make sure we don't break on potentially new versions (this package is deprecated, but security patches are always possible)
- `2.2.0` is the 2nd most used, if I would have picked the 1st most used, I would have tested 2.2.5 twice
- `2.0.4` is the last stable since 2.0 intro - it could have been 2.1.38 as well